File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.java
import java.util.*;

class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int n = nums.size();

        for (int i = 0; i <= n - 2 * k; i++) {
            List<Integer> arr1 = nums.subList(i, i + k);
            List<Integer> arr2 = nums.subList(i + k, i + 2 * k);

            if (isStrictlyIncreasing(arr1) && isStrictlyIncreasing(arr2)) {
                return true;
            }
        }
        return false;
    }

    private boolean isStrictlyIncreasing(List<Integer> arr) {
        for (int i = 1; i < arr.size(); i++) {
            if (arr.get(i) <= arr.get(i - 1)) return false;
        }
        return true;
    }
}

File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        for i in range(n - 2 * k + 1):
            arr1 = nums[i:i+k]
            arr2 = nums[i+k:i+2*k]
            
            if self.isStrictlyIncreasing(arr1) and self.isStrictlyIncreasing(arr2):
                return True
        return False

    def isStrictlyIncreasing(self, arr):
        for i in range(1, len(arr)):
            if arr[i] <= arr[i-1]:
                return False
        return True
